bentoml/yatai/client/bento_repository_api.py

```diff
@@ -274,12 +356,34 @@ def _update_bento_upload_progress(
         self.yatai_service.UpdateBento(update_bento_req)
 
     def download_to_directory(self, bento_pb, target_dir):
-        if bento_pb.uri.s3_presigned_url:
-            bento_service_bundle_path = bento_pb.uri.s3_presigned_url
-        elif bento_pb.uri.gcs_presigned_url:
-            bento_service_bundle_path = bento_pb.uri.gcs_presigned_url
+        """
+        Download or move bundle bundle to target directory
+
+        Args:
+            bento_pb: bento bundle protobuf dict
+            target_dir: path
+
+        Returns:
+            None
+        """
+        if bento_pb.uri.type == BentoUri.S3 or bento_pb.uri.type == BentoUri.GCS:
+            bento_service_bundle_path = (
```

Comment:
same as above, prefer specific multi-line if-else - which makes it a lot easier when we need to add support or a new blob storage service.